### ISSUE TITLE
Failed install on Windows

### ISSUE THREAD

**Issue by duxingzhe** [tensorflow/tensorflow#16478]
Python 3.6.4

There is a strange error when I install tensorflow 1.5.

```
Collecting tensorflow
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/34/96/11f048eca7b4d6da3084ca49c636b9e720e9dd1483c0c4e9ba3cf5037564/tensorflow-1.5.0-cp36-cp36m-win_amd64.whl
Requirement already up-to-date: wheel>=0.26 in d:\python\python36\lib\site-packages (from tensorflow)
Requirement already up-to-date: numpy>=1.12.1 in d:\python\python36\lib\site-packages (from tensorflow)
Collecting absl-py>=0.1.6 (from tensorflow)
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/42/3c/1985d86a44bfe44fd060c02807336f840a509bfaa2d340860fba7d22da39/absl-py-0.1.9.tar.gz
Requirement already up-to-date: protobuf>=3.4.0 in d:\python\python36\lib\site-packages (from tensorflow)
Requirement already up-to-date: six>=1.10.0 in d:\python\python36\lib\site-packages (from tensorflow)
Collecting tensorflow-tensorboard<1.6.0,>=1.5.0 (from tensorflow)
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/43/69/82e2a368076c94edbba3cd15804103bf1f31486d69e11551b71fa1d1f384/tensorflow_tensorboard-1.5.0-py3-none-any.whl
Requirement already up-to-date: setuptools in d:\python\python36\lib\site-packages (from protobuf>=3.4.0->tensorflow)
Requirement already up-to-date: bleach==1.5.0 in d:\python\python36\lib\site-packages (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow)
Requirement already up-to-date: markdown>=2.6.8 in d:\python\python36\lib\site-packages (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow)
Requirement already up-to-date: werkzeug>=0.11.10 in d:\python\python36\lib\site-packages (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow)
Requirement already up-to-date: html5lib==0.9999999 in d:\python\python36\lib\site-packages (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow)
Collecting futures>=3.1.1 (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow)
  Using cached https://pypi.tuna.tsinghua.edu.cn/packages/1f/9e/7b2ff7e965fc654592269f2906ade1c7d705f1bf25b7d469fa153f7d19eb/futures-3.2.0.tar.gz
Unknown requires Python '>=2.6, <3' but the running Python is 3.6.4
```

Why the dependency is *futures*? It doesn't have a verion of Python 3.6.4.

**Comment by tatianashp** [comment]
It is hard to understand your problem without complete information.  What installation method do you use?

**Comment by fpsandnoob** [comment]
I have a same problem on the Ubuntu 16.04. I use pip to get the Python 3.5 binary file. And it is strange that Python 3 doesn't need the futures package.

**Comment by tatianashp** [comment]
@fpsandnoob Could you please specify what method you use to install TensorFlow?

**Comment by fpsandnoob** [comment]
@tatianashp I used the native pip to install tensorflow-gpu with the version of Python 3.5. When the pip install the dependency, the problem occurred. Like this one:
`pip install --ignore-installed --upgrade https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow_gpu-1.5.0-cp35-cp35m-linux_x86_64.whl`

**Comment by WongChen** [comment]
Same problem here on Ubuntu 16.04 with Python 3.5.
Clearly, Python 3 do not need futures.  And with Python 2.7, using : 
`pip2 install tensorflow-gpu==1.5.0`
it goes well.

**Comment by duxingzhe** [comment]
Thanks for previous solution.

**Comment by pokey** [comment]
I think this issue should be reopened.  @WongChen has posted a workaround, but this is still a bug in tensorflow's dependency setup, is it not?

**Comment by HustCoderHu** [comment]
agree with pokey, this is a bug

**Comment by zsrkmyn** [comment]
agree with @pokey, this is actully a bug.

**Comment by dguny** [comment]
i'd argue this is more a problem with futures, not with TF since TF's requirement with futures is 3.1.1+ which works fine on Python3.6

**Comment by WongChen** [comment]
I think the problem here is that futures > 3.1.1 is not allowed to be installed for py3.
see [here](https://pypi.python.org/pypi/futures).
Before we can install tensorflow for py3 with futures installed, but now since pip install the newest version, the error is occurred.
If you google it, you can find many errors just like this.
On the other side, I think tf for py3 package's dependency **should not include futures**.

**Comment by dguny** [comment]
ah, my bad. i assumed that it was ok to install futures 3.1.1 for python 3 but seems like they just forgot to check the version.

**Comment by tatianashp** [comment]
@chihuahua Could you please take a look at tensorboard dependency on futures?

**Comment by yaox12** [comment]
I think the problem is caused by tensorboard dependency, it requires futures >= 3.1.1.
> Collecting futures>=3.1.1 (from tensorflow-tensorboard<1.6.0,>=1.5.0->tensorflow)  
    futures requires Python '>=2.6, <3' but the running Python is 3.6.4

**Comment by tatianashp** [comment]
@yaox12 You are correct. It is tensorboard issue. See tensorflow/tensorboard/#916. tensorflow/tensorboard/#920 PR fixes the problem.

**Comment by songsh** [comment]
pip3 install futures==3.1.1
pip3 install tensorflow==1.14.0  not work

**Comment by mihaimaruseac** [comment]
Please open new issue, this is for Tf 1.5 which is no longer supported.

Locking conversation since this issue has been solved in 2018.
